parser.y

```diff
@@ -4020,6 +4051,209 @@ ExplainFormatType:
+		stmt.Kind = ast.BRIEKindRestore
+		stmt.Storage = $5
+		stmt.Options = $6.([]*ast.BRIEOption)
+		if $3 != nil {
+			stmt.Incremental = $3.(*ast.BRIEOption)
+		}
+		$$ = stmt
+	}
+
+BRIETables:
+	DatabaseSym '*'
+	{
+		$$ = &ast.BRIEStmt{}
+	}
+|	DatabaseSym DBNameList
+	{
+		$$ = &ast.BRIEStmt{Schemas: $2.([]string)}
+	}
+|	"TABLE" TableNameList
+	{
+		$$ = &ast.BRIEStmt{Tables: $2.([]*ast.TableName)}
+	}
+
+DBNameList:
+	DBName
+	{
+		$$ = []string{$1.(string)}
+	}
+|	DBNameList ',' DBName
+	{
+		$$ = append($1.([]string), $3.(string))
+	}
+
+BackupTypeSpec:
+	%prec empty
+	{
+		$$ = nil
+	}
+|	"FULL"
+	{
+		$$ = nil
+	}
+|	"INCREMENTAL" "UNTIL" "TIMESTAMP" stringLit
+	{
+		$$ = &ast.BRIEOption{
+			Tp:       ast.BRIEOptionLastBackupTS,
+			StrValue: $4,
+		}
+	}
+|	"INCREMENTAL" "UNTIL" "TIMESTAMP_ORACLE" LengthNum
+	{
+		$$ = &ast.BRIEOption{
+			Tp:        ast.BRIEOptionLastBackupTSO,
+			UintValue: $4.(uint64),
+		}
+	}
+
+BRIEOptions:
+	%prec empty
+	{
+		$$ = []*ast.BRIEOption{}
+	}
+|	BRIEOptions BRIEOption
+	{
+		$$ = append($1.([]*ast.BRIEOption), $2.(*ast.BRIEOption))
+	}
+
+BRIEIntegerOptionName:
+	"CONCURRENCY"
+	{
+		$$ = ast.BRIEOptionConcurrency
+	}
+|	"CHECKSUM"
+	{
+		$$ = ast.BRIEOptionChecksum
+	}
+|	"SEND_CREDENTIALS_TO_TIKV"
+	{
+		$$ = ast.BRIEOptionSendCreds
+	}
+|	"ONLINE"
+	{
+		$$ = ast.BRIEOptionOnline
+	}
+|	"S3_FORCE_PATH_STYLE"
+	{
+		$$ = ast.BRIEOptionS3ForcePathStyle
+	}
+|	"S3_USE_ACCELERATE_ENDPOINT"
+	{
+		$$ = ast.BRIEOptionS3UseAccelerateEndpoint
+	}
+
+BRIEStringOptionName:
+	"S3_ENDPOINT"
+	{
+		$$ = ast.BRIEOptionS3Endpoint
+	}
+|	"S3_REGION"
+	{
+		$$ = ast.BRIEOptionS3Region
+	}
+|	"S3_STORAGE_CLASS"
+	{
+		$$ = ast.BRIEOptionS3StorageClass
+	}
+|	"S3_SSE"
+	{
+		$$ = ast.BRIEOptionS3SSE
+	}
+|	"S3_ACL"
+	{
+		$$ = ast.BRIEOptionS3ACL
+	}
+|	"S3_ACCESS_KEY"
+	{
+		$$ = ast.BRIEOptionS3AccessKey
+	}
+|	"S3_SECRET_ACCESS_KEY"
+	{
+		$$ = ast.BRIEOptionS3SecretAccessKey
+	}
+|	"S3_PROVIDER"
+	{
+		$$ = ast.BRIEOptionS3Provider
+	}
+|	"GCS_ENDPOINT"
+	{
+		$$ = ast.BRIEOptionGCSEndpoint
+	}
+|	"GCS_STORAGE_CLASS"
+	{
+		$$ = ast.BRIEOptionGCSStorageClass
+	}
+|	"GCS_PREDEFINED_ACL"
+	{
+		$$ = ast.BRIEOptionGCSPredefinedACL
+	}
+|	"GCS_CREDENTIALS_FILE"
+	{
+		$$ = ast.BRIEOptionGCSCredentialsFile
+	}
+
+BRIEOption:
+	BRIEIntegerOptionName EqOpt LengthNum
+	{
+		$$ = &ast.BRIEOption{
+			Tp:        $1.(ast.BRIEOptionType),
+			UintValue: $3.(uint64),
+		}
+	}
+|	BRIEStringOptionName EqOpt stringLit
+	{
+		$$ = &ast.BRIEOption{
+			Tp:       $1.(ast.BRIEOptionType),
+			StrValue: $3,
+		}
+	}
+|	"SNAPSHOT" EqOpt LengthNum TimestampUnit "AGO"
+	{
+		unit, err := $4.(ast.TimeUnitType).Duration()
+		if err != nil {
+			yylex.AppendError(err)
+			return 1
+		}
+		// TODO: check overflow?
+		$$ = &ast.BRIEOption{
+			Tp:        ast.BRIEOptionBackupTimeAgo,
+			UintValue: $3.(uint64) * uint64(unit),
+		}
+	}
+|	"RATE_LIMIT" EqOpt LengthNum "MB" '/' "SECOND"
+	{
+		// TODO: check overflow?
```

Comment:
How can it overflow :grey_question:

Reply:
@tangenta `RATE_LIMIT = 9223372036854775807 MB/SECOND`... admittedly this is not a real issue 😅